client/src/App.js

```diff
@@ -0,0 +1,222 @@
+
+export const MeetupContext = React.createContext();
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiKey: '',
+      textbox: '',
+      customFields: [],
+      checked: []
+    };
+  }
+
+  setKey = (event) => {
+    this.setState({textbox: event.target.value});
+  }
+
+  handleKeyPress = (event) => {
+    if(event.key === 'Enter'){
+      this.handleFind(event);
+    }
+  }
+
+  handleFind = (e) => {
+    e.preventDefault();
+    if (this.state.textbox) {
+      this.setState({apiKey : this.state.textbox}, () => {
+        this.fetchFields();
+      });
+    }
+  }
+
+  fetchSavedSettings = () => {
+    axios.post('http://localhost:5000/api/saved-fields', {
+        apiKey: this.state.apiKey
+      })
+    .then(response => {
+      let checked = this.state.checked || [];
+      let checkbox;
+      let x;
+      response.data.forEach( (savedField) => {
+        checkbox = checked.find(({name})=> name === savedField);
+
+        if (!checkbox) {
+          checked.push({name: savedField, checked: true});
+          x = document.getElementById(`${savedField}`);
+          x.checked = true;
+        } else {
+          checkbox.checked = true;
+        }
+
+        this.setState({ checked });
+      })
+    })
+    .catch(err => {
+      alert('Could not fetch retrieve saved settings. Please check your API key.');
+      console.log(err, 'Could not retrieve settings');
+    });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    // only store the field if checked === true
+    let checkedFields = this.state.checked.filter(({checked})=> checked).map(({name})=>name);
+
+    axios.post('http://localhost:5000/api/settings', {
+        fieldsToSave: checkedFields,
+        apiKey: this.state.apiKey
+      })
+    .then(response => {
+      alert('Settings saved.');
+    })
+    .catch(err => {
+      console.log(err, 'Could not save settings');
+      alert('There was a problem. Settings could not be saved.');
+    });
+  }
+
+  handleCheckboxChange = (e) => {
+    // dynamically update state object when checkbox is clicked/unclicked
+    let checked = this.state.checked || [];
+    let checkbox = checked.find(({name})=> name === e.target.id);
+
+    if (!checkbox) {
+      checked.push({name: e.target.id, checked: e.target.checked});
+    } else {
+      checkbox.checked = e.target.checked;
+    }
+
+    this.setState({ checked })
+  }
+
+  fetchFields = () => {
+
+    axios.post('http://localhost:5000/api/custom-fields', {
+        apiKey: this.state.apiKey
+      })
+      .catch(err => {
+        console.log(err, 'Could not fetch custom fields');
+      })
+      .then(response => {
+        if (response && response.data) {
+          this.setState({customFields: response.data});
+          response.data.forEach( (customField) => {
+            this.setState({ [customField.replace(/ /g, "")]: false });
+          })
+        }
+      this.fetchSavedSettings();
+    })
+  }
```

Comment:
indentation is... interesting here, making intention unclear: `this.fetchSavedSettings();` happens inside the `.then` but it doesn't appear that way.